### ISSUE TITLE
(5.0.12,5.0.11) Default NoteSkin is overridden by alphabetical name

### ISSUE THREAD

**Issue by cgoudie** [stepmania/stepmania#1261]
MacOS-X El Capitan. Case insensitive filesystem.

In 5.0.12, If you add some new NoteSkins, such as DDX-RAINBOW, the default will be changed from "default" to "ddx-rainbow" on first start.

If you go into the settings, and attempt to change it from this back to default and click save, the result is not saved. The arrows are not changed and viewing the setting again shows "ddx-rainbow"

If you quit, go back to the filesystem, and rename the folders from DDX-RAINBOW to zzDDX-RAINBOW default will return as the default.

**Comment by kyzentun** [comment]
Are you sure the theme isn't overriding it with the DefaultNoteSkinName metric?

**Comment by 1033Forest** [comment]
The same issue has happened to me too on 5.0.11. I had a bunch of noteskins, it always decides to use the first one in alphabetical order as the default. So for me, in this case, 3dnote is default

**Comment by cgoudie** [comment]
Pretty sure. I poked about in the metrics file, and the word default isn't inside. It does specify a FallbackNoteSkin. 

Also, if this was the case, it is suspicious that the alphabetical order would impact the default. Finally, I would think it would still allow me to save default as the option, which it doesn't.

**Comment by kyzentun** [comment]
Before we continue, did you delete the noteskin named "default"?  Removing that tends to cause problems.

The DefaultNoteSkin "preference" that you pick in the Appearance Options section is actually stored as part of the DefaultModifiers string.  During loading and when set from a menu, the modifier string is crushed to lower case.  When the modifier string is actually saved to the prefs file, the first letter of each part is capitalized.

The preference menu picks the first option in the row if it can't figure out which option is selected.

So maybe the crushing and capitalizing is distorting the noteskin name you're picking as default, causing problems with saving the option.  What happens if you give it an all lowercase name?

**Comment by cgoudie** [comment]
@kyzentun No, I didn't delete the default noteskin, as I would prefer the default stay my default :).
